diff -Nur smalltalk-1.95.9/configure.ac smalltalk-1.95.9.new/configure.ac
--- smalltalk-1.95.9/configure.ac	Tue Jan 15 19:26:39 2002
+++ smalltalk-1.95.9.new/configure.ac	Mon Jan 21 14:24:40 2002
@@ -15,7 +15,7 @@
 dnl CURRENT:REVISION:AGE means this is the REVISION-th version of
 dnl the CURRENT-th interface; all the interface from CURRENT-AGE
 dnl to CURRENT are supported.
-AC_INIT([GNU Smalltalk], 2.0.6, help-smalltalk@gnu.org, smalltalk)
+AC_INIT([GNU Smalltalk], 2.0.6, help-smalltalk@gnu.org, gnu-smalltalk)
 GST_REVISION(3:6:0)
 AC_CONFIG_AUX_DIR([config])
 AC_CONFIG_SRCDIR([main.c])
