diff -Naur smalltalk-1.7/Makefile.body smalltalk-1.7-patched/Makefile.body
--- smalltalk-1.7/Makefile.body	Mon Feb 14 08:11:30 2000
+++ smalltalk-1.7-patched/Makefile.body	Sat Feb 19 21:01:28 2000
@@ -25,196 +25,197 @@
 MAIN_SOURCES = $(srcdir)/main.c $(srcdir)/cfuncs.c
 
 # Many thanks to find, tr and fmt for help in producing these lists!
-KERNEL_FILES = $(srcdir)/kernel/Array.st $(srcdir)/kernel/ArrayColl.st \
-	$(srcdir)/kernel/Association.st $(srcdir)/kernel/Autoload.st \
-	$(srcdir)/kernel/CharArray.st $(srcdir)/kernel/Bag.st \
-	$(srcdir)/kernel/Behavior.st $(srcdir)/kernel/BlkClosure.st \
-	$(srcdir)/kernel/BlkContext.st $(srcdir)/kernel/Boolean.st \
-	$(srcdir)/kernel/Browser.st $(srcdir)/kernel/Builtins.st \
-	$(srcdir)/kernel/ByteArray.st $(srcdir)/kernel/ByteMemory.st \
-	$(srcdir)/kernel/ByteStream.st $(srcdir)/kernel/CFuncs.st \
-	$(srcdir)/kernel/CObject.st $(srcdir)/kernel/CStruct.st \
-	$(srcdir)/kernel/CType.st $(srcdir)/kernel/Character.st \
-	$(srcdir)/kernel/Class.st $(srcdir)/kernel/ClassDesc.st \
-	$(srcdir)/kernel/Collection.st $(srcdir)/kernel/CompildCode.st \
-	$(srcdir)/kernel/CompildMeth.st $(srcdir)/kernel/CompiledBlk.st \
-	$(srcdir)/kernel/ContextPart.st $(srcdir)/kernel/Date.st \
-	$(srcdir)/kernel/Delay.st $(srcdir)/kernel/Dictionary.st \
-	$(srcdir)/kernel/DirMessage.st $(srcdir)/kernel/Directory.st \
-	$(srcdir)/kernel/DLD.st $(srcdir)/kernel/ExcHandling.st \
-	$(srcdir)/kernel/False.st $(srcdir)/kernel/File.st \
-	$(srcdir)/kernel/FileSegment.st $(srcdir)/kernel/FileStream.st \
-	$(srcdir)/kernel/Float.st $(srcdir)/kernel/Fraction.st \
-	$(srcdir)/kernel/IdentDict.st $(srcdir)/kernel/IdentitySet.st \
-	$(srcdir)/kernel/Integer.st $(srcdir)/kernel/Interval.st \
-	$(srcdir)/kernel/LargeInt.st $(srcdir)/kernel/Link.st \
-	$(srcdir)/kernel/LinkedList.st $(srcdir)/kernel/LookupKey.st \
-	$(srcdir)/kernel/LookupTable.st $(srcdir)/kernel/Magnitude.st \
-	$(srcdir)/kernel/MappedColl.st $(srcdir)/kernel/Memory.st \
-	$(srcdir)/kernel/Message.st $(srcdir)/kernel/Metaclass.st \
-	$(srcdir)/kernel/MethodInfo.st $(srcdir)/kernel/MethodDict.st \
-	$(srcdir)/kernel/MthContext.st $(srcdir)/kernel/Namespace.st \
-	$(srcdir)/kernel/Number.st $(srcdir)/kernel/ObjDumper.st \
-	$(srcdir)/kernel/Object.st $(srcdir)/kernel/OrderColl.st \
-	$(srcdir)/kernel/PkgLoader.st $(srcdir)/kernel/Point.st \
-	$(srcdir)/kernel/PosStream.st $(srcdir)/kernel/ProcSched.st \
-	$(srcdir)/kernel/Process.st $(srcdir)/kernel/RWStream.st \
-	$(srcdir)/kernel/Random.st $(srcdir)/kernel/ReadStream.st \
-	$(srcdir)/kernel/Rectangle.st $(srcdir)/kernel/RootNamespc.st \
-	$(srcdir)/kernel/RunArray.st $(srcdir)/kernel/Semaphore.st \
-	$(srcdir)/kernel/SeqCollect.st $(srcdir)/kernel/Set.st \
-	$(srcdir)/kernel/SharedQueue.st $(srcdir)/kernel/SmallInt.st \
-	$(srcdir)/kernel/SortCollect.st \
-	$(srcdir)/kernel/Stream.st $(srcdir)/kernel/String.st \
-	$(srcdir)/kernel/SymLink.st $(srcdir)/kernel/Symbol.st \
-	$(srcdir)/kernel/SysDict.st $(srcdir)/kernel/Time.st \
-	$(srcdir)/kernel/TokenStream.st $(srcdir)/kernel/Transcript.st \
-	$(srcdir)/kernel/True.st $(srcdir)/kernel/UndefObject.st \
-	$(srcdir)/kernel/ValueAdapt.st $(srcdir)/kernel/WeakObjects.st \
-	$(srcdir)/kernel/WordMemory.st $(srcdir)/kernel/WriteStream.st \
-	$(srcdir)/kernel/initialize.st
+KERNEL_FILES = kernel/Array.st kernel/ArrayColl.st \
+	kernel/Association.st kernel/Autoload.st \
+	kernel/CharArray.st kernel/Bag.st \
+	kernel/Behavior.st kernel/BlkClosure.st \
+	kernel/BlkContext.st kernel/Boolean.st \
+	kernel/Browser.st kernel/Builtins.st \
+	kernel/ByteArray.st kernel/ByteMemory.st \
+	kernel/ByteStream.st kernel/CFuncs.st \
+	kernel/CObject.st kernel/CStruct.st \
+	kernel/CType.st kernel/Character.st \
+	kernel/Class.st kernel/ClassDesc.st \
+	kernel/Collection.st kernel/CompildCode.st \
+	kernel/CompildMeth.st kernel/CompiledBlk.st \
+	kernel/ContextPart.st kernel/Date.st \
+	kernel/Delay.st kernel/Dictionary.st \
+	kernel/DirMessage.st kernel/Directory.st \
+	kernel/DLD.st kernel/ExcHandling.st \
+	kernel/False.st kernel/File.st \
+	kernel/FileSegment.st kernel/FileStream.st \
+	kernel/Float.st kernel/Fraction.st \
+	kernel/IdentDict.st kernel/IdentitySet.st \
+	kernel/Integer.st kernel/Interval.st \
+	kernel/LargeInt.st kernel/Link.st \
+	kernel/LinkedList.st kernel/LookupKey.st \
+	kernel/LookupTable.st kernel/Magnitude.st \
+	kernel/MappedColl.st kernel/Memory.st \
+	kernel/Message.st kernel/Metaclass.st \
+	kernel/MethodInfo.st kernel/MethodDict.st \
+	kernel/MthContext.st kernel/Namespace.st \
+	kernel/Number.st kernel/ObjDumper.st \
+	kernel/Object.st kernel/OrderColl.st \
+	kernel/PkgLoader.st kernel/Point.st \
+	kernel/PosStream.st kernel/ProcSched.st \
+	kernel/Process.st kernel/RWStream.st \
+	kernel/Random.st kernel/ReadStream.st \
+	kernel/Rectangle.st kernel/RootNamespc.st \
+	kernel/RunArray.st kernel/Semaphore.st \
+	kernel/SeqCollect.st kernel/Set.st \
+	kernel/SharedQueue.st kernel/SmallInt.st \
+	kernel/SortCollect.st \
+	kernel/Stream.st kernel/String.st \
+	kernel/SymLink.st kernel/Symbol.st \
+	kernel/SysDict.st kernel/Time.st \
+	kernel/TokenStream.st kernel/Transcript.st \
+	kernel/True.st kernel/UndefObject.st \
+	kernel/ValueAdapt.st kernel/WeakObjects.st \
+	kernel/WordMemory.st kernel/WriteStream.st \
+	kernel/initialize.st
 
 MAIN_ST_SOURCES = \
-	$(srcdir)/Load.st $(srcdir)/Reload.st $(KERNEL_FILES) \
-	$(srcdir)/cint/CDeclNode.st $(srcdir)/cint/CExpressionNode.st \
-	$(srcdir)/cint/CPP.st $(srcdir)/cint/CPStrConc.st \
-	$(srcdir)/cint/CPStrUnq.st $(srcdir)/cint/CParseType.st \
-	$(srcdir)/cint/CSymbol.st $(srcdir)/cint/CSymbolTable.st \
-	$(srcdir)/cint/CTok.st $(srcdir)/cint/CToken.st \
-	$(srcdir)/cint/CTreeBuilder.st $(srcdir)/cint/CollapsedStream.st \
-	$(srcdir)/cint/CPStrUnq.st $(srcdir)/cint/CPStrConc.st \
-	$(srcdir)/cint/README $(srcdir)/cint/ExpansionStream.st \
-	$(srcdir)/cint/ExpansionStreamStack.st $(srcdir)/cint/LineStream.st \
-	$(srcdir)/cint/LineTokenStream.st $(srcdir)/cint/StreamStack.st \
-	$(srcdir)/compiler/Load.st $(srcdir)/compiler/STCompLit.st \
-	$(srcdir)/compiler/StartCompiler.st $(srcdir)/compiler/STFileInParser.st \
-	$(srcdir)/compiler/STParseNodes.st $(srcdir)/compiler/STParser.st \
-	$(srcdir)/compiler/STSymTable.st $(srcdir)/compiler/STTok.st \
-	$(srcdir)/compiler/STToken.st $(srcdir)/compiler/README
-
-GDBM_FILES = $(srcdir)/gdbm/gdbm.st $(srcdir)/gdbm/Makefile.body \
-	$(srcdir)/gdbm/cfuncs.h
-
-EXAMPLE_FILES = $(srcdir)/examples/Application.st $(srcdir)/examples/Bench.st \
-	$(srcdir)/examples/Case.st $(srcdir)/examples/CompFileSeg.st \
-	$(srcdir)/examples/DBFile.st $(srcdir)/examples/Debugger.st \
-	$(srcdir)/examples/EditStream.st $(srcdir)/examples/GenClasses.st \
-	$(srcdir)/examples/IndStream.st $(srcdir)/examples/MemUsage.st \
-	$(srcdir)/examples/CStream.st $(srcdir)/examples/Dinner.st \
-	$(srcdir)/examples/PrtHier.st $(srcdir)/examples/Publish.st \
-	$(srcdir)/examples/Queens.st $(srcdir)/examples/PushBack.st \
-	$(srcdir)/examples/RandomInt.st $(srcdir)/examples/Buffers.st \
-	$(srcdir)/examples/README $(srcdir)/examples/RecursLit.st \
-	$(srcdir)/examples/RegExp.st $(srcdir)/examples/Tokenizer.st \
-	$(srcdir)/examples/Richards.st $(srcdir)/examples/Timer.st \
-	$(srcdir)/examples/shell $(srcdir)/examples/Watchdog.st \
-	$(srcdir)/examples/Prolog.st $(srcdir)/examples/Lisp.st \
-	$(srcdir)/examples/blox/Man.st $(srcdir)/examples/blox/Tetris.st \
-	$(srcdir)/examples/mixed/self.st $(srcdir)/examples/mixed/resolve.st \
-	$(srcdir)/examples/mixed/a.st $(srcdir)/examples/mixed/ba.st \
-	$(srcdir)/examples/mixed/BigExpr.st $(srcdir)/examples/mixed/bug.st \
-	$(srcdir)/examples/mixed/bug2.st $(srcdir)/examples/mixed/bug4.st \
-	$(srcdir)/examples/mixed/CInt.st $(srcdir)/examples/mixed/binary.c \
-	$(srcdir)/examples/mixed/Compiler.st \
-	$(srcdir)/examples/mixed/CParseT1.st \
-	$(srcdir)/examples/mixed/CParseT2.st \
-	$(srcdir)/examples/mixed/CParseT3.st \
-	$(srcdir)/examples/mixed/CPP-test.st \
-	$(srcdir)/examples/mixed/CString.st $(srcdir)/examples/mixed/DLD.st \
-	$(srcdir)/examples/mixed/market.st $(srcdir)/examples/mixed/er2.st \
-	$(srcdir)/examples/mixed/Extensn.st \
-	$(srcdir)/examples/mixed/IfError.st \
-	$(srcdir)/examples/mixed/Interr.st \
-	$(srcdir)/examples/mixed/ParseErr.st \
-	$(srcdir)/examples/mixed/Process1.st \
-	$(srcdir)/examples/mixed/Process2.st \
-	$(srcdir)/examples/mixed/Profile.st \
-	$(srcdir)/examples/mixed/Sorting.st \
-	$(srcdir)/examples/mixed/t.st $(srcdir)/examples/mixed/torture.st
-
-LIBRARY_FILES =	$(srcdir)/lib/Makefile.in $(srcdir)/lib/alloc.c \
-	$(srcdir)/lib/alloc.h $(srcdir)/lib/alloca.c $(srcdir)/lib/byte.c \
-	$(srcdir)/lib/byte.h $(srcdir)/lib/callin.c $(srcdir)/lib/callin.h \
-	$(srcdir)/lib/cfuncs.c $(srcdir)/lib/cint.c $(srcdir)/lib/cint.h \
-	$(srcdir)/lib/comp.c $(srcdir)/lib/comp.h $(srcdir)/lib/dict.c \
-	$(srcdir)/lib/dict.h $(srcdir)/lib/dict.inl $(srcdir)/lib/dynamic.c \
-	$(srcdir)/lib/dynamic.h $(srcdir)/lib/dld_aix.c \
-	$(srcdir)/lib/dld_gnu.c $(srcdir)/lib/dld_gst.c \
-	$(srcdir)/lib/dld_gst.h $(srcdir)/lib/dld_hpux.c \
-	$(srcdir)/lib/dld_libdl.c $(srcdir)/lib/dld_ltdl.c \
-	$(srcdir)/lib/dld_none.c $(srcdir)/lib/dld_win32.c \
-	$(srcdir)/lib/gst.h $(srcdir)/lib/gst.tab.c \
-	$(srcdir)/lib/gst.tab.h $(srcdir)/lib/gst.y $(srcdir)/lib/gstpub.h \
-	$(srcdir)/lib/interp.c $(srcdir)/lib/interp.h \
-	$(srcdir)/lib/interp.inl $(srcdir)/lib/lex.c $(srcdir)/lib/lex.h \
-	$(srcdir)/lib/lib.c $(srcdir)/lib/lib.h $(srcdir)/lib/memzero.h \
-	$(srcdir)/lib/obstack.c $(srcdir)/lib/obstack.h $(srcdir)/lib/oop.c \
-	$(srcdir)/lib/oop.h $(srcdir)/lib/oop.inl $(srcdir)/lib/prims.inl \
-	$(srcdir)/lib/putenv.c $(srcdir)/lib/qsort.c $(srcdir)/lib/register.h \
-	$(srcdir)/lib/save.c $(srcdir)/lib/save.h $(srcdir)/lib/str.c \
-	$(srcdir)/lib/str.h $(srcdir)/lib/strdup.c $(srcdir)/lib/strtoul.c \
-	$(srcdir)/lib/sym.c $(srcdir)/lib/sym.h $(srcdir)/lib/sysdep.c \
-	$(srcdir)/lib/sysdep.h $(srcdir)/lib/tree.c $(srcdir)/lib/tree.h \
-	$(srcdir)/lib/waitpid.c
-
-DOC_FILES = $(srcdir)/docs/COPYING $(srcdir)/docs/COPYING.LIB $(srcdir)/docs/categories \
-	$(srcdir)/docs/NEWS $(srcdir)/docs/todo $(srcdir)/docs/tutorial.txi \
-	$(srcdir)/docs/gpl.txi $(srcdir)/docs/gst.txi $(srcdir)/docs/toc.txi \
-	$(srcdir)/docs/ChangeLog-TCP $(srcdir)/docs/ChangeLog-Blox \
-	$(srcdir)/docs/ChangeLog-C $(srcdir)/docs/ChangeLog-ST \
-	$(srcdir)/docs/Makefile $(srcdir)/docs/texinfo.tex \
-	$(srcdir)/docs/gst.info* $(srcdir)/docs/DOM.html \
-	$(srcdir)/docs/stamp-classes $(srcdir)/docs/AUTHORS
-
-EMACS_FILES = $(srcdir)/emacs/bison.el $(srcdir)/emacs/st-changelog.el $(srcdir)/emacs/st.el
-
-TEST_FILES = $(srcdir)/test/arrays.ok $(srcdir)/test/arrays.st \
-	$(srcdir)/test/blocks.ok $(srcdir)/test/blocks.st \
-	$(srcdir)/test/chars.ok $(srcdir)/test/chars.st \
-	$(srcdir)/test/classes.ok $(srcdir)/test/classes.st \
-	$(srcdir)/test/dates.ok $(srcdir)/test/dates.st \
-	$(srcdir)/test/delays.ok $(srcdir)/test/delays.st \
-	$(srcdir)/test/do-check \
-	$(srcdir)/test/exceptions.ok $(srcdir)/test/exceptions.st \
-	$(srcdir)/test/floatmath.ok $(srcdir)/test/floatmath.st \
-	$(srcdir)/test/geometry.ok $(srcdir)/test/geometry.st \
-	$(srcdir)/test/intmath.ok $(srcdir)/test/intmath.st \
-	$(srcdir)/test/objects.ok $(srcdir)/test/objects.st \
-	$(srcdir)/test/processes.ok $(srcdir)/test/processes.st \
-	$(srcdir)/test/sets.ok $(srcdir)/test/sets.st \
-	$(srcdir)/test/strings.ok $(srcdir)/test/strings.st
-
-BLOX_FILES = $(srcdir)/blox/BBrowser.st $(srcdir)/blox/BB-Run.st \
-	$(srcdir)/blox/bear.gif $(srcdir)/blox/Colors.txt \
-	$(srcdir)/blox/BrowserMain.st $(srcdir)/blox/ClassBrow.st \
-	$(srcdir)/blox/CtxtInspect.st $(srcdir)/blox/GuiData.st \
-	$(srcdir)/blox/DictInspect.st $(srcdir)/blox/MethInspect.st \
-	$(srcdir)/blox/Notifier.st $(srcdir)/blox/Load.st \
-	$(srcdir)/blox/Run.st $(srcdir)/blox/BrowShell.st \
-	$(srcdir)/blox/Inspector.st $(srcdir)/blox/Manager.st \
-	$(srcdir)/blox/Menu.st $(srcdir)/blox/MethSetBrow.st \
-	$(srcdir)/blox/ModalDialog.st $(srcdir)/blox/PList.st \
-	$(srcdir)/blox/PText.st $(srcdir)/blox/RadioForm.st \
-	$(srcdir)/blox/StrcInspect.st $(srcdir)/blox/PCode.st \
-	$(srcdir)/blox/View.st $(srcdir)/blox/Blox.st \
-	$(srcdir)/blox/BloxBasic.st $(srcdir)/blox/BloxCanvas.st \
-	$(srcdir)/blox/BloxExtend.st $(srcdir)/blox/BloxText.st \
-	$(srcdir)/blox/BloxWidgets.st $(srcdir)/blox/test.st \
-	$(srcdir)/blox/Blox.c $(srcdir)/blox/cfuncs.h \
-	$(srcdir)/blox/Makefile.body $(srcdir)/blox/rgbtab.h
-
-TCP_FILES = $(srcdir)/tcp/AbstractSocket.st $(srcdir)/tcp/init.st \
-	$(srcdir)/tcp/AbstractSocketImpl.st $(srcdir)/tcp/cfuncs.st \
-	$(srcdir)/tcp/ContentHandler.st $(srcdir)/tcp/Datagram.st \
-	$(srcdir)/tcp/DatagramSocket.st $(srcdir)/tcp/DatagramSocketImpl.st \
-	$(srcdir)/tcp/IPAddress.st $(srcdir)/tcp/MulticastSocket.st \
-	$(srcdir)/tcp/ServerSocket.st $(srcdir)/tcp/Socket.st \
-	$(srcdir)/tcp/SocketImpl.st $(srcdir)/tcp/TCP.st \
-	$(srcdir)/tcp/TCPSocketImpl.st $(srcdir)/tcp/UDPSocketImpl.st \
-	$(srcdir)/tcp/UnixStream.st $(srcdir)/tcp/URL.st \
-	$(srcdir)/tcp/tcp.c $(srcdir)/tcp/Makefile.body $(srcdir)/tcp/cfuncs.h
-
-XML_FILES = $(srcdir)/xml/DOM.st $(srcdir)/xml/XWalker.st
+	Load.st Reload.st $(KERNEL_FILES) \
+	cint/CDeclNode.st cint/CExpressionNode.st \
+	cint/CPP.st cint/CPStrConc.st \
+	cint/CPStrUnq.st cint/CParseType.st \
+	cint/CSymbol.st cint/CSymbolTable.st \
+	cint/CTok.st cint/CToken.st \
+	cint/CTreeBuilder.st cint/CollapsedStream.st \
+	cint/CPStrUnq.st cint/CPStrConc.st \
+	cint/README cint/ExpansionStream.st \
+	cint/ExpansionStreamStack.st cint/LineStream.st \
+	cint/LineTokenStream.st cint/StreamStack.st \
+	compiler/Load.st compiler/STCompLit.st \
+	compiler/StartCompiler.st compiler/STVisitor.st \
+	compiler/STParseNodes.st compiler/STParser.st \
+	compiler/STSymTable.st compiler/STTok.st \
+	compiler/STToken.st compiler/STCompiler.st \
+	compiler/Readme
+
+GDBM_FILES = examples/gdbm/gdbm.st examples/gdbm/Makefile.body \
+	examples/gdbm/cfuncs.h
+
+EXAMPLE_FILES = examples/Application.st examples/Bench.st \
+	examples/Case.st examples/CompFileSeg.st \
+	examples/DBFile.st examples/Debugger.st \
+	examples/EditStream.st examples/GenClasses.st \
+	examples/IndStream.st examples/MemUsage.st \
+	examples/CStream.st examples/Dinner.st \
+	examples/PrtHier.st examples/Publish.st \
+	examples/Queens.st examples/PushBack.st \
+	examples/RandomInt.st examples/Buffers.st \
+	examples/README examples/RecursLit.st \
+	examples/RegExp.st examples/Tokenizer.st \
+	examples/Richards.st examples/Timer.st \
+	examples/shell examples/Watchdog.st \
+	examples/Prolog.st examples/Lisp.st \
+	examples/blox/Man.st examples/blox/Tetris.st \
+	examples/mixed/self.st examples/mixed/resolve.st \
+	examples/mixed/a.st examples/mixed/ba.st \
+	examples/mixed/BigExpr.st examples/mixed/bug.st \
+	examples/mixed/bug2.st examples/mixed/bug4.st \
+	examples/mixed/CInt.st examples/mixed/binary.c \
+	examples/mixed/Compiler.st \
+	examples/mixed/CParseT1.st \
+	examples/mixed/CParseT2.st \
+	examples/mixed/CParseT3.st \
+	examples/mixed/CPP-test.st \
+	examples/mixed/CString.st examples/mixed/DLD.st \
+	examples/mixed/market.st examples/mixed/er2.st \
+	examples/mixed/Extensn.st \
+	examples/mixed/IfError.st \
+	examples/mixed/Interr.st \
+	examples/mixed/ParseErr.st \
+	examples/mixed/Process1.st \
+	examples/mixed/Process2.st \
+	examples/mixed/Profile.st \
+	examples/mixed/Sorting.st \
+	examples/mixed/t.st examples/mixed/torture.st
+
+LIBRARY_FILES = lib/Makefile.in lib/alloc.c \
+	lib/alloc.h lib/alloca.c lib/byte.c \
+	lib/byte.h lib/callin.c lib/callin.h \
+	lib/cfuncs.c lib/cint.c lib/cint.h \
+	lib/comp.c lib/comp.h lib/dict.c \
+	lib/dict.h lib/dict.inl lib/dynamic.c \
+	lib/dynamic.h lib/dld_aix.c \
+	lib/dld_gnu.c lib/dld_gst.c \
+	lib/dld_gst.h lib/dld_hpux.c \
+	lib/dld_libdl.c lib/dld_ltdl.c \
+	lib/dld_none.c lib/dld_win32.c \
+	lib/gst.h lib/gst.tab.c \
+	lib/gst.tab.h lib/gst.y lib/gstpub.h \
+	lib/interp.c lib/interp.h \
+	lib/interp.inl lib/lex.c lib/lex.h \
+	lib/lib.c lib/lib.h lib/memzero.h \
+	lib/obstack.c lib/obstack.h lib/oop.c \
+	lib/oop.h lib/oop.inl lib/prims.inl \
+	lib/putenv.c lib/qsort.c lib/register.h \
+	lib/save.c lib/save.h lib/str.c \
+	lib/str.h lib/strdup.c lib/strtoul.c \
+	lib/sym.c lib/sym.h lib/sysdep.c \
+	lib/sysdep.h lib/tree.c lib/tree.h \
+	lib/waitpid.c
+
+DOC_FILES = docs/COPYING docs/COPYING.LIB docs/categories \
+	docs/NEWS docs/todo docs/tutorial.txi \
+	docs/gpl.txi docs/gst.txi docs/toc.txi \
+	docs/ChangeLog-TCP docs/ChangeLog-Blox \
+	docs/ChangeLog-C docs/ChangeLog-ST \
+	docs/Makefile docs/texinfo.tex \
+	docs/gst.info* docs/DOM.html \
+	docs/stamp-classes docs/AUTHORS
+
+EMACS_FILES = emacs/bison.el emacs/st-changelog.el emacs/st.el
+
+TEST_FILES = test/arrays.ok test/arrays.st \
+	test/blocks.ok test/blocks.st \
+	test/chars.ok test/chars.st \
+	test/classes.ok test/classes.st \
+	test/dates.ok test/dates.st \
+	test/delays.ok test/delays.st \
+	test/do-check \
+	test/exceptions.ok test/exceptions.st \
+	test/floatmath.ok test/floatmath.st \
+	test/geometry.ok test/geometry.st \
+	test/intmath.ok test/intmath.st \
+	test/objects.ok test/objects.st \
+	test/processes.ok test/processes.st \
+	test/sets.ok test/sets.st \
+	test/strings.ok test/strings.st
+
+BLOX_FILES = blox/BBrowser.st blox/BB-Run.st \
+	blox/bear.gif blox/Colors.txt \
+	blox/BrowserMain.st blox/ClassBrow.st \
+	blox/CtxtInspect.st blox/GuiData.st \
+	blox/DictInspect.st blox/MethInspect.st \
+	blox/Notifier.st blox/Load.st \
+	blox/Run.st blox/BrowShell.st \
+	blox/Inspector.st blox/Manager.st \
+	blox/Menu.st blox/MethSetBrow.st \
+	blox/ModalDialog.st blox/PList.st \
+	blox/PText.st blox/RadioForm.st \
+	blox/StrcInspect.st blox/PCode.st \
+	blox/View.st blox/Blox.st \
+	blox/BloxBasic.st blox/BloxCanvas.st \
+	blox/BloxExtend.st blox/BloxText.st \
+	blox/BloxWidgets.st blox/test.st \
+	blox/Blox.c blox/cfuncs.h \
+	blox/Makefile.body blox/rgbtab.h
+
+TCP_FILES = tcp/AbstractSocket.st tcp/Init.st \
+	tcp/AbstractSocketImpl.st tcp/cfuncs.st \
+	tcp/ContentHandler.st tcp/Datagram.st \
+	tcp/DatagramSocket.st tcp/DatagramSocketImpl.st \
+	tcp/IPAddress.st tcp/MulticastSocket.st \
+	tcp/ServerSocket.st tcp/Socket.st \
+	tcp/SocketImpl.st tcp/TCP.st \
+	tcp/TCPSocketImpl.st tcp/UDPSocketImpl.st \
+	tcp/UnixStream.st tcp/URL.st \
+	tcp/tcp.c tcp/Makefile.body tcp/cfuncs.h
+ 
+XML_FILES = xml/DOM.st xml/XWalker.st
 
 M4_FILES = m4/waitpid.m4 m4/dld_api.m4 m4/double.m4 m4/ext_goto.m4 \
 	m4/inet.m4 m4/localtime.m4 m4/obstack.m4 m4/oldansi.m4 \
@@ -235,12 +236,14 @@
 OTHER_FILES = gsticon.ico INSTALL install-pkg packages
 
 INSTALLED_FILES = $(GDBM_FILES) $(MAIN_ST_SOURCES) $(TEST_FILES) \
-	$(TCP_FILES) $(EXAMPLE_FILES) $(BLOX_FILES)
+	$(TCP_FILES) $(EXAMPLE_FILES) $(BLOX_FILES) $(XML_FILES)
 
-INSTALLED_DIRS = $(stdir) $(stdir)/kernel $(stdir)/blox $(stdir)/gdbm \
-	$(stdir)/examples $(stdir)/examples/blox $(stdir)/examples/mixed \
-	$(stdir)/tcp $(stdir)/cint $(stdir)/compiler $(stdir)/test \
-	$(stdir)/xml
+INSTALLED_DIRS = $(DESTDIR)$(stdir) $(DESTDIR)$(stdir)/kernel \
+	$(DESTDIR)$(stdir)/blox $(DESTDIR)$(stdir)/examples \
+	$(DESTDIR)$(stdir)/examples/blox $(DESTDIR)$(stdir)/examples/mixed \
+	$(DESTDIR)$(stdir)/tcp $(DESTDIR)$(stdir)/cint \
+	$(DESTDIR)$(stdir)/compiler $(DESTDIR)$(stdir)/test \
+	$(DESTDIR)$(stdir)/examples/gdbm $(DESTDIR)$(stdir)/xml
 
 OBJECTS = main.o cfuncs.o
 
diff -Naur smalltalk-1.7/Makefile.body.orig smalltalk-1.7-patched/Makefile.body.orig
--- smalltalk-1.7/Makefile.body.orig	Thu Jan  1 01:00:00 1970
+++ smalltalk-1.7-patched/Makefile.body.orig	Sat Feb 19 20:59:55 2000
@@ -0,0 +1,251 @@
+# Master Makefile for GNU Smalltalk
+# Copyright 1988-92, 1994-95, 1999, 2000 Free Software Foundation, Inc.
+# Written by Steve Byrne
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+
+# $Revision$
+
+# Make sure this is the first target
+all: $(PROGS)
+
+
+MAIN_SOURCES = $(srcdir)/main.c $(srcdir)/cfuncs.c
+
+# Many thanks to find, tr and fmt for help in producing these lists!
+KERNEL_FILES = $(srcdir)/kernel/Array.st $(srcdir)/kernel/ArrayColl.st \
+	$(srcdir)/kernel/Association.st $(srcdir)/kernel/Autoload.st \
+	$(srcdir)/kernel/CharArray.st $(srcdir)/kernel/Bag.st \
+	$(srcdir)/kernel/Behavior.st $(srcdir)/kernel/BlkClosure.st \
+	$(srcdir)/kernel/BlkContext.st $(srcdir)/kernel/Boolean.st \
+	$(srcdir)/kernel/Browser.st $(srcdir)/kernel/Builtins.st \
+	$(srcdir)/kernel/ByteArray.st $(srcdir)/kernel/ByteMemory.st \
+	$(srcdir)/kernel/ByteStream.st $(srcdir)/kernel/CFuncs.st \
+	$(srcdir)/kernel/CObject.st $(srcdir)/kernel/CStruct.st \
+	$(srcdir)/kernel/CType.st $(srcdir)/kernel/Character.st \
+	$(srcdir)/kernel/Class.st $(srcdir)/kernel/ClassDesc.st \
+	$(srcdir)/kernel/Collection.st $(srcdir)/kernel/CompildCode.st \
+	$(srcdir)/kernel/CompildMeth.st $(srcdir)/kernel/CompiledBlk.st \
+	$(srcdir)/kernel/ContextPart.st $(srcdir)/kernel/Date.st \
+	$(srcdir)/kernel/Delay.st $(srcdir)/kernel/Dictionary.st \
+	$(srcdir)/kernel/DirMessage.st $(srcdir)/kernel/Directory.st \
+	$(srcdir)/kernel/DLD.st $(srcdir)/kernel/ExcHandling.st \
+	$(srcdir)/kernel/False.st $(srcdir)/kernel/File.st \
+	$(srcdir)/kernel/FileSegment.st $(srcdir)/kernel/FileStream.st \
+	$(srcdir)/kernel/Float.st $(srcdir)/kernel/Fraction.st \
+	$(srcdir)/kernel/IdentDict.st $(srcdir)/kernel/IdentitySet.st \
+	$(srcdir)/kernel/Integer.st $(srcdir)/kernel/Interval.st \
+	$(srcdir)/kernel/LargeInt.st $(srcdir)/kernel/Link.st \
+	$(srcdir)/kernel/LinkedList.st $(srcdir)/kernel/LookupKey.st \
+	$(srcdir)/kernel/LookupTable.st $(srcdir)/kernel/Magnitude.st \
+	$(srcdir)/kernel/MappedColl.st $(srcdir)/kernel/Memory.st \
+	$(srcdir)/kernel/Message.st $(srcdir)/kernel/Metaclass.st \
+	$(srcdir)/kernel/MethodInfo.st $(srcdir)/kernel/MethodDict.st \
+	$(srcdir)/kernel/MthContext.st $(srcdir)/kernel/Namespace.st \
+	$(srcdir)/kernel/Number.st $(srcdir)/kernel/ObjDumper.st \
+	$(srcdir)/kernel/Object.st $(srcdir)/kernel/OrderColl.st \
+	$(srcdir)/kernel/PkgLoader.st $(srcdir)/kernel/Point.st \
+	$(srcdir)/kernel/PosStream.st $(srcdir)/kernel/ProcSched.st \
+	$(srcdir)/kernel/Process.st $(srcdir)/kernel/RWStream.st \
+	$(srcdir)/kernel/Random.st $(srcdir)/kernel/ReadStream.st \
+	$(srcdir)/kernel/Rectangle.st $(srcdir)/kernel/RootNamespc.st \
+	$(srcdir)/kernel/RunArray.st $(srcdir)/kernel/Semaphore.st \
+	$(srcdir)/kernel/SeqCollect.st $(srcdir)/kernel/Set.st \
+	$(srcdir)/kernel/SharedQueue.st $(srcdir)/kernel/SmallInt.st \
+	$(srcdir)/kernel/SortCollect.st \
+	$(srcdir)/kernel/Stream.st $(srcdir)/kernel/String.st \
+	$(srcdir)/kernel/SymLink.st $(srcdir)/kernel/Symbol.st \
+	$(srcdir)/kernel/SysDict.st $(srcdir)/kernel/Time.st \
+	$(srcdir)/kernel/TokenStream.st $(srcdir)/kernel/Transcript.st \
+	$(srcdir)/kernel/True.st $(srcdir)/kernel/UndefObject.st \
+	$(srcdir)/kernel/ValueAdapt.st $(srcdir)/kernel/WeakObjects.st \
+	$(srcdir)/kernel/WordMemory.st $(srcdir)/kernel/WriteStream.st \
+	$(srcdir)/kernel/initialize.st
+
+MAIN_ST_SOURCES = \
+	$(srcdir)/Load.st $(srcdir)/Reload.st $(KERNEL_FILES) \
+	$(srcdir)/cint/CDeclNode.st $(srcdir)/cint/CExpressionNode.st \
+	$(srcdir)/cint/CPP.st $(srcdir)/cint/CPStrConc.st \
+	$(srcdir)/cint/CPStrUnq.st $(srcdir)/cint/CParseType.st \
+	$(srcdir)/cint/CSymbol.st $(srcdir)/cint/CSymbolTable.st \
+	$(srcdir)/cint/CTok.st $(srcdir)/cint/CToken.st \
+	$(srcdir)/cint/CTreeBuilder.st $(srcdir)/cint/CollapsedStream.st \
+	$(srcdir)/cint/CPStrUnq.st $(srcdir)/cint/CPStrConc.st \
+	$(srcdir)/cint/README $(srcdir)/cint/ExpansionStream.st \
+	$(srcdir)/cint/ExpansionStreamStack.st $(srcdir)/cint/LineStream.st \
+	$(srcdir)/cint/LineTokenStream.st $(srcdir)/cint/StreamStack.st \
+	$(srcdir)/compiler/Load.st $(srcdir)/compiler/STCompLit.st \
+	$(srcdir)/compiler/StartCompiler.st $(srcdir)/compiler/STFileInParser.st \
+	$(srcdir)/compiler/STParseNodes.st $(srcdir)/compiler/STParser.st \
+	$(srcdir)/compiler/STSymTable.st $(srcdir)/compiler/STTok.st \
+	$(srcdir)/compiler/STToken.st $(srcdir)/compiler/README
+
+GDBM_FILES = $(srcdir)/gdbm/gdbm.st $(srcdir)/gdbm/Makefile.body \
+	$(srcdir)/gdbm/cfuncs.h
+
+EXAMPLE_FILES = $(srcdir)/examples/Application.st $(srcdir)/examples/Bench.st \
+	$(srcdir)/examples/Case.st $(srcdir)/examples/CompFileSeg.st \
+	$(srcdir)/examples/DBFile.st $(srcdir)/examples/Debugger.st \
+	$(srcdir)/examples/EditStream.st $(srcdir)/examples/GenClasses.st \
+	$(srcdir)/examples/IndStream.st $(srcdir)/examples/MemUsage.st \
+	$(srcdir)/examples/CStream.st $(srcdir)/examples/Dinner.st \
+	$(srcdir)/examples/PrtHier.st $(srcdir)/examples/Publish.st \
+	$(srcdir)/examples/Queens.st $(srcdir)/examples/PushBack.st \
+	$(srcdir)/examples/RandomInt.st $(srcdir)/examples/Buffers.st \
+	$(srcdir)/examples/README $(srcdir)/examples/RecursLit.st \
+	$(srcdir)/examples/RegExp.st $(srcdir)/examples/Tokenizer.st \
+	$(srcdir)/examples/Richards.st $(srcdir)/examples/Timer.st \
+	$(srcdir)/examples/shell $(srcdir)/examples/Watchdog.st \
+	$(srcdir)/examples/Prolog.st $(srcdir)/examples/Lisp.st \
+	$(srcdir)/examples/blox/Man.st $(srcdir)/examples/blox/Tetris.st \
+	$(srcdir)/examples/mixed/self.st $(srcdir)/examples/mixed/resolve.st \
+	$(srcdir)/examples/mixed/a.st $(srcdir)/examples/mixed/ba.st \
+	$(srcdir)/examples/mixed/BigExpr.st $(srcdir)/examples/mixed/bug.st \
+	$(srcdir)/examples/mixed/bug2.st $(srcdir)/examples/mixed/bug4.st \
+	$(srcdir)/examples/mixed/CInt.st $(srcdir)/examples/mixed/binary.c \
+	$(srcdir)/examples/mixed/Compiler.st \
+	$(srcdir)/examples/mixed/CParseT1.st \
+	$(srcdir)/examples/mixed/CParseT2.st \
+	$(srcdir)/examples/mixed/CParseT3.st \
+	$(srcdir)/examples/mixed/CPP-test.st \
+	$(srcdir)/examples/mixed/CString.st $(srcdir)/examples/mixed/DLD.st \
+	$(srcdir)/examples/mixed/market.st $(srcdir)/examples/mixed/er2.st \
+	$(srcdir)/examples/mixed/Extensn.st \
+	$(srcdir)/examples/mixed/IfError.st \
+	$(srcdir)/examples/mixed/Interr.st \
+	$(srcdir)/examples/mixed/ParseErr.st \
+	$(srcdir)/examples/mixed/Process1.st \
+	$(srcdir)/examples/mixed/Process2.st \
+	$(srcdir)/examples/mixed/Profile.st \
+	$(srcdir)/examples/mixed/Sorting.st \
+	$(srcdir)/examples/mixed/t.st $(srcdir)/examples/mixed/torture.st
+
+LIBRARY_FILES =	$(srcdir)/lib/Makefile.in $(srcdir)/lib/alloc.c \
+	$(srcdir)/lib/alloc.h $(srcdir)/lib/alloca.c $(srcdir)/lib/byte.c \
+	$(srcdir)/lib/byte.h $(srcdir)/lib/callin.c $(srcdir)/lib/callin.h \
+	$(srcdir)/lib/cfuncs.c $(srcdir)/lib/cint.c $(srcdir)/lib/cint.h \
+	$(srcdir)/lib/comp.c $(srcdir)/lib/comp.h $(srcdir)/lib/dict.c \
+	$(srcdir)/lib/dict.h $(srcdir)/lib/dict.inl $(srcdir)/lib/dynamic.c \
+	$(srcdir)/lib/dynamic.h $(srcdir)/lib/dld_aix.c \
+	$(srcdir)/lib/dld_gnu.c $(srcdir)/lib/dld_gst.c \
+	$(srcdir)/lib/dld_gst.h $(srcdir)/lib/dld_hpux.c \
+	$(srcdir)/lib/dld_libdl.c $(srcdir)/lib/dld_ltdl.c \
+	$(srcdir)/lib/dld_none.c $(srcdir)/lib/dld_win32.c \
+	$(srcdir)/lib/gst.h $(srcdir)/lib/gst.tab.c \
+	$(srcdir)/lib/gst.tab.h $(srcdir)/lib/gst.y $(srcdir)/lib/gstpub.h \
+	$(srcdir)/lib/interp.c $(srcdir)/lib/interp.h \
+	$(srcdir)/lib/interp.inl $(srcdir)/lib/lex.c $(srcdir)/lib/lex.h \
+	$(srcdir)/lib/lib.c $(srcdir)/lib/lib.h $(srcdir)/lib/memzero.h \
+	$(srcdir)/lib/obstack.c $(srcdir)/lib/obstack.h $(srcdir)/lib/oop.c \
+	$(srcdir)/lib/oop.h $(srcdir)/lib/oop.inl $(srcdir)/lib/prims.inl \
+	$(srcdir)/lib/putenv.c $(srcdir)/lib/qsort.c $(srcdir)/lib/register.h \
+	$(srcdir)/lib/save.c $(srcdir)/lib/save.h $(srcdir)/lib/str.c \
+	$(srcdir)/lib/str.h $(srcdir)/lib/strdup.c $(srcdir)/lib/strtoul.c \
+	$(srcdir)/lib/sym.c $(srcdir)/lib/sym.h $(srcdir)/lib/sysdep.c \
+	$(srcdir)/lib/sysdep.h $(srcdir)/lib/tree.c $(srcdir)/lib/tree.h \
+	$(srcdir)/lib/waitpid.c
+
+DOC_FILES = $(srcdir)/docs/COPYING $(srcdir)/docs/COPYING.LIB $(srcdir)/docs/categories \
+	$(srcdir)/docs/NEWS $(srcdir)/docs/todo $(srcdir)/docs/tutorial.txi \
+	$(srcdir)/docs/gpl.txi $(srcdir)/docs/gst.txi $(srcdir)/docs/toc.txi \
+	$(srcdir)/docs/ChangeLog-TCP $(srcdir)/docs/ChangeLog-Blox \
+	$(srcdir)/docs/ChangeLog-C $(srcdir)/docs/ChangeLog-ST \
+	$(srcdir)/docs/Makefile $(srcdir)/docs/texinfo.tex \
+	$(srcdir)/docs/gst.info* $(srcdir)/docs/DOM.html \
+	$(srcdir)/docs/stamp-classes $(srcdir)/docs/AUTHORS
+
+EMACS_FILES = $(srcdir)/emacs/bison.el $(srcdir)/emacs/st-changelog.el $(srcdir)/emacs/st.el
+
+TEST_FILES = $(srcdir)/test/arrays.ok $(srcdir)/test/arrays.st \
+	$(srcdir)/test/blocks.ok $(srcdir)/test/blocks.st \
+	$(srcdir)/test/chars.ok $(srcdir)/test/chars.st \
+	$(srcdir)/test/classes.ok $(srcdir)/test/classes.st \
+	$(srcdir)/test/dates.ok $(srcdir)/test/dates.st \
+	$(srcdir)/test/delays.ok $(srcdir)/test/delays.st \
+	$(srcdir)/test/do-check \
+	$(srcdir)/test/exceptions.ok $(srcdir)/test/exceptions.st \
+	$(srcdir)/test/floatmath.ok $(srcdir)/test/floatmath.st \
+	$(srcdir)/test/geometry.ok $(srcdir)/test/geometry.st \
+	$(srcdir)/test/intmath.ok $(srcdir)/test/intmath.st \
+	$(srcdir)/test/objects.ok $(srcdir)/test/objects.st \
+	$(srcdir)/test/processes.ok $(srcdir)/test/processes.st \
+	$(srcdir)/test/sets.ok $(srcdir)/test/sets.st \
+	$(srcdir)/test/strings.ok $(srcdir)/test/strings.st
+
+BLOX_FILES = $(srcdir)/blox/BBrowser.st $(srcdir)/blox/BB-Run.st \
+	$(srcdir)/blox/bear.gif $(srcdir)/blox/Colors.txt \
+	$(srcdir)/blox/BrowserMain.st $(srcdir)/blox/ClassBrow.st \
+	$(srcdir)/blox/CtxtInspect.st $(srcdir)/blox/GuiData.st \
+	$(srcdir)/blox/DictInspect.st $(srcdir)/blox/MethInspect.st \
+	$(srcdir)/blox/Notifier.st $(srcdir)/blox/Load.st \
+	$(srcdir)/blox/Run.st $(srcdir)/blox/BrowShell.st \
+	$(srcdir)/blox/Inspector.st $(srcdir)/blox/Manager.st \
+	$(srcdir)/blox/Menu.st $(srcdir)/blox/MethSetBrow.st \
+	$(srcdir)/blox/ModalDialog.st $(srcdir)/blox/PList.st \
+	$(srcdir)/blox/PText.st $(srcdir)/blox/RadioForm.st \
+	$(srcdir)/blox/StrcInspect.st $(srcdir)/blox/PCode.st \
+	$(srcdir)/blox/View.st $(srcdir)/blox/Blox.st \
+	$(srcdir)/blox/BloxBasic.st $(srcdir)/blox/BloxCanvas.st \
+	$(srcdir)/blox/BloxExtend.st $(srcdir)/blox/BloxText.st \
+	$(srcdir)/blox/BloxWidgets.st $(srcdir)/blox/test.st \
+	$(srcdir)/blox/Blox.c $(srcdir)/blox/cfuncs.h \
+	$(srcdir)/blox/Makefile.body $(srcdir)/blox/rgbtab.h
+
+TCP_FILES = $(srcdir)/tcp/AbstractSocket.st $(srcdir)/tcp/init.st \
+	$(srcdir)/tcp/AbstractSocketImpl.st $(srcdir)/tcp/cfuncs.st \
+	$(srcdir)/tcp/ContentHandler.st $(srcdir)/tcp/Datagram.st \
+	$(srcdir)/tcp/DatagramSocket.st $(srcdir)/tcp/DatagramSocketImpl.st \
+	$(srcdir)/tcp/IPAddress.st $(srcdir)/tcp/MulticastSocket.st \
+	$(srcdir)/tcp/ServerSocket.st $(srcdir)/tcp/Socket.st \
+	$(srcdir)/tcp/SocketImpl.st $(srcdir)/tcp/TCP.st \
+	$(srcdir)/tcp/TCPSocketImpl.st $(srcdir)/tcp/UDPSocketImpl.st \
+	$(srcdir)/tcp/UnixStream.st $(srcdir)/tcp/URL.st \
+	$(srcdir)/tcp/tcp.c $(srcdir)/tcp/Makefile.body $(srcdir)/tcp/cfuncs.h
+
+XML_FILES = $(srcdir)/xml/DOM.st $(srcdir)/xml/XWalker.st
+
+M4_FILES = m4/waitpid.m4 m4/dld_api.m4 m4/double.m4 m4/ext_goto.m4 \
+	m4/inet.m4 m4/localtime.m4 m4/obstack.m4 m4/oldansi.m4 \
+	m4/readline.m4 m4/signed.m4 m4/strtoul.m4 m4/symbolset.m4 \
+	m4/tcltk.m4 m4/checkprog.m4
+
+CONFIGURE_FILES = acconfig.h configure configure.in gstconf.h.in install-pkg \
+	Makefile.body Makefile.defs.in Makefile.rules modules short-names \
+	config.guess config.sub stamp-h.in install-sh dosconf.h \
+	dos-config.cache $(M4_FILES)
+
+CONFIGURE_GENERATED = gstconf.h Makefile.defs lib/Makefile
+
+DOS_REMOVE = *.in lib/*.in Makefile.* config.status dos-config.cache \
+	config.status config.cache install-sh stamp-h short-names \
+	blox examples/blox tcp
+
+OTHER_FILES = gsticon.ico INSTALL install-pkg packages
+
+INSTALLED_FILES = $(GDBM_FILES) $(MAIN_ST_SOURCES) $(TEST_FILES) \
+	$(TCP_FILES) $(EXAMPLE_FILES) $(BLOX_FILES)
+
+INSTALLED_DIRS = $(stdir) $(stdir)/kernel $(stdir)/blox $(stdir)/gdbm \
+	$(stdir)/examples $(stdir)/examples/blox $(stdir)/examples/mixed \
+	$(stdir)/tcp $(stdir)/cint $(stdir)/compiler $(stdir)/test \
+	$(stdir)/xml
+
+OBJECTS = main.o cfuncs.o
+
+SOURCES = $(MAIN_SOURCES)
+
+DISTFILES = $(MAIN_SOURCES) $(CONFIGURE_FILES) $(OTHER_FILES) \
+	$(LIBRARY_FILES) $(DOC_FILES) $(EMACS_FILES) $(MAIN_ST_SOURCES) \
+	$(INSTALLED_FILES) $(XML_FILES)
diff -Naur smalltalk-1.7/Makefile.rules smalltalk-1.7-patched/Makefile.rules
--- smalltalk-1.7/Makefile.rules	Mon Feb 14 08:11:31 2000
+++ smalltalk-1.7-patched/Makefile.rules	Sat Feb 19 21:21:13 2000
@@ -12,7 +12,7 @@
 ######################################################################
 
 
-.PHONY : mostlyclean clean distclean maintainer-clean attempt all dist_dir
+.PHONY : mostlyclean clean distclean maintainer-clean attempt all
 
 ########################### BUILD RULES ##############################
 
@@ -41,7 +41,7 @@
 gsticon.o: gsticon.ico
 	echo ProgramIcon ICON gsticon.ico | windres -o gsticon.o
 
-$(srcdir)/lib/libgst.a: $(LIBRARY_FILES) $(srcdir)/gstconf.h
+$(srcdir)/lib/libgst.a: $(LIBRARY_FILES)
 	(cd lib; $(MAKE) CFLAGS="$(CFLAGS)" GCC_CFLAGS="$(GCC_CFLAGS)" \
 			 CDEBUG="$(CDEBUG)" SCHEDULER="$(SCHEDULER)")
 
@@ -57,13 +57,13 @@
 # config.status might not change gstconf.h
 gstconf.h: stamp-h
 stamp-h: gstconf.h.in config.status
-	cd $(srcdir) && ./config.status
+#	cd $(srcdir) && ./config.status
 
 lib/Makefile.in: lib/Makefile config.status
-	cd $(srcdir) && ./config.status
+#	cd $(srcdir) && ./config.status
 
 Makefile.defs.in: Makefile.defs config.status
-	cd $(srcdir) && ./config.status
+#	cd $(srcdir) && ./config.status
 
 configure: configure.in aclocal.m4
 	cd $(srcdir) && $(AUTOCONF)
@@ -103,23 +103,25 @@
 	(cd docs; $(MAKE) dvi; cd ..)
 
 installdirs:
-	$(SHELL) $(srcdir)/mkinstalldirs $(bindir) $(smlibdir) $(infodir) \
-		 $(includedir) $(man1dir) $(INSTALLED_DIRS)
-
-install:: installdirs optimize
-	$(INSTALL_PROGRAM) gst $(bindir)/gst
-	$(INSTALL_DATA) lib/gst.h $(includedir)/gst.h
-	-$(INSTALL_DATA) lib/gst.h $(oldincludedir)/gst.h
-	$(INSTALL_DATA) gstconf.h $(includedir)/gstconf.h
-	-$(INSTALL_DATA) gstconf.h $(oldincludedir)/gstconf.h
-	$(INSTALL_DATA) lib/gstpub.h $(includedir)/gstpub.h
-	-$(INSTALL_DATA) lib/gstpub.h $(oldincludedir)/gstpub.h
-	$(INSTALL_DATA) lib/libgst.a $(smlibdir)/libgst.a
-	(cd $(srcdir)/docs; $(MAKE) install; cd ..)
+	$(SHELL) $(srcdir)/mkinstalldirs $(DESTDIR)$(bindir) \
+		$(DESTDIR)$(smlibdir) $(DESTDIR)$(infodir) \
+		$(DESTDIR)$(includedir) $(DESTDIR)$(man1dir) \
+		$(INSTALLED_DIRS)
+
+install:: installdirs
+	$(INSTALL_PROGRAM) gst $(DESTDIR)$(bindir)/gst
+	$(INSTALL_DATA) lib/gst.h $(DESTDIR)$(includedir)/gst.h
+	-$(INSTALL_DATA) lib/gst.h $(DESTDIR)$(oldincludedir)/gst.h
+	$(INSTALL_DATA) gstconf.h $(DESTDIR)$(includedir)/gstconf.h
+	-$(INSTALL_DATA) gstconf.h $(DESTDIR)$(oldincludedir)/gstconf.h
+	$(INSTALL_DATA) lib/gstpub.h $(DESTDIR)$(includedir)/gstpub.h
+	-$(INSTALL_DATA) lib/gstpub.h $(DESTDIR)$(oldincludedir)/gstpub.h
+	$(INSTALL_DATA) lib/libgst.a $(DESTDIR)$(smlibdir)/libgst.a
+	(cd $(srcdir)/docs; $(MAKE) DESTDIR=$(DESTDIR) install; cd ..)
 	for file in $(INSTALLED_FILES); do				\
-	  $(INSTALL_DATA) $$file $(stdir)/$$file;			\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(stdir)/$$file;	\
 	done
-	(x=`pwd`; cd $(IMAGE_PATH); $$x/gst dummyFile -Vi; cd $$x)
+	(x=`pwd`; cd $(DESTDIR)$(IMAGE_PATH); $$x/gst dummyFile -Vi; cd $$x)
 
 uninstall:
 	$(RM) $(bindir)/gst		$(smlibdir)/libgst.a		\
diff -Naur smalltalk-1.7/configure.in smalltalk-1.7-patched/configure.in
--- smalltalk-1.7/configure.in	Mon Feb 14 08:11:29 2000
+++ smalltalk-1.7-patched/configure.in	Sat Feb 19 20:59:55 2000
@@ -155,8 +155,10 @@
 
 ### HACK!
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
-AC_DEFINE_UNQUOTED(KERNEL_PATH, "`eval echo $datadir`/gnu-smalltalk/kernel")
-AC_DEFINE_UNQUOTED(IMAGE_PATH, "`eval echo $datadir`/gnu-smalltalk")
+KERNEL_PATH="`eval echo $datadir`/gnu-smalltalk/kernel"
+IMAGE_PATH="`eval echo $datadir`/gnu-smalltalk"
+AC_DEFINE_UNQUOTED(KERNEL_PATH, "$KERNEL_PATH")
+AC_DEFINE_UNQUOTED(IMAGE_PATH, "$IMAGE_PATH")
 
 AC_SUBST(CDEBUG)
 AC_SUBST(CFLAGS)
diff -Naur smalltalk-1.7/docs/Makefile smalltalk-1.7-patched/docs/Makefile
--- smalltalk-1.7/docs/Makefile	Mon Feb 14 08:11:46 2000
+++ smalltalk-1.7-patched/docs/Makefile	Sat Feb 19 21:26:22 2000
@@ -18,7 +18,7 @@
 
 # $Revision$
 
-include ../Makefile.body
+include ../Makefile.defs
 
 .PHONY : mostlyclean clean distclean maintainer-clean do-mantainer-clean all info dvi
 
@@ -48,24 +48,24 @@
 	-(cd $(srcdir)/docs; $(TEXI2DVI) $<)
 
 # Using a timestamp file allows me to avoid distributing classes.txi
-classes.txi: $(KERNEL_FILES)
+classes.txi: 
 	cd ..;								\
 	echo ClassPublisher Texinfo publishAllDocs! |			\
 	  ./gst examples/Publish.st - -qg
 	echo timestamp>stamp-classes
 
-stamp-classes: $(KERNEL_FILES)
+stamp-classes: 
 	$(MAKE) classes.txi
 
 # Install targets
 install: info
-	$(INSTALL_DATA) gst.1 $(mandir)/man1/gst.1
+	$(INSTALL_DATA) gst.1 $(DESTDIR)$(mandir)/man1/gst.1
 	cd $(srcdir)/docs; for file in $(INFO)*; do			\
-	  $(INSTALL_DATA) $$file $(infodir);				\
-	done
-	cd $(srcdir)/docs; for file in $(INFO); do			\
-	  $(INSTALL_INFO) --dir-file=$(infodir)/dir $(infodir)/$$file	\
+	  $(INSTALL_DATA) $$file $(DESTDIR)$(infodir);				\
 	done
+#	cd $(srcdir)/docs; for file in $(INFO); do			\
+#	  $(INSTALL_INFO) --dir-file=$(infodir)/dir $(infodir)/$$file	\
+#	done
 
 uninstall:
 	-$(RM) $(mandir)/man1/gst.1
