--- smalltalk-1.95.1/kernel/Makefile.am.wiget	Fri Apr 13 20:42:44 2001
+++ smalltalk-1.95.1/kernel/Makefile.am	Fri Apr 13 20:53:34 2001
@@ -32,6 +32,6 @@
 # Build an image after installing the data; install-data runs after
 # install-exec, so the gst executable is already in bindir.
 
-install-data-local:
-	cd $(basedir) && $(bindir)/gst -Vi dummyFile
+#install-data-local: 
+#	cd $(basedir) && $(bindir)/gst -Vi dummyFile
 
